service-statistics-unsd/src/main/java/org/oskari/statistics/plugins/unsd/UnsdRequest.java

```diff
@@ -151,6 +151,8 @@ private String getData(String path, Map<String, String> params) throws APIExcept
         try {
             final String url = getUrl(path, params);
             con = IOHelper.getConnection(url);
+            // 30 seconds as this might take a while and we are _usually_ doing this on the background thread
+            con.setReadTimeout(30000);
```

Comment:
If there's a lot of data we could add a `Accept-Encoding: gzip` header here and call `IOHelper.readString(HttpURLConnection)` instead.

Reply:
I'll add this as a configuration option as well. The response isn't huge but for some reason connection to the API can be a bit shaky at times.